tools/extract_metadata.py

```diff
@@ -241,13 +241,10 @@ def get_main_reads_params(module, export_map):
 
 def get_global_exports(module, exports):
   global_exports = {}
-  internal_start_stop_symbols = set(['__start_em_asm', '__stop_em_asm',
-                                     '__start_em_lib_deps', '__stop_em_lib_deps',
-                                     '__em_lib_deps'])
   internal_prefixes = ('__em_js__', '__em_lib_deps')
   for export in exports:
     if export.kind == webassembly.ExternType.GLOBAL:
-      if export.name in internal_start_stop_symbols or any(export.name.startswith(p) for p in internal_prefixes):
+      if any(export.name.startswith(p) for p in internal_prefixes):
```

Comment:
How is this part of the change related?

Reply:
This is needed since we use the set of global exports here:

https://github.com/emscripten-core/emscripten/blob/7f369e1e9f27af3385471bb00314a73d467d72d6/tools/building.py#L751-L752

Without this the list of unused exports we pass also includes some of these globals, which would then cause the assertion to fire.

Since this is the only place the `WASM_GLOBAL_EXPORTS` is used I think its fine.

Reply:
I factored out `is_internal_global` here which I think makes things more clear.